api/envoy/api/v2/cds.proto

```diff
@@ -442,6 +442,14 @@ message Cluster {
   //   to an upstream host that becomes unhealthy, Envoy may spend a substantial amount of
   //   time exclusively closing these connections, and not processing any other traffic.
   bool close_connections_on_host_health_failure = 31;
+
+  // If this cluster uses EDS and its ClusterLoadAssignment is updated such that some endpoints
+  // have been removed, immediately drain connections and remove the endpoint from the cluster.
+  //
+  // This only affects behavior for endpoints that are being health checked. If this flag
```

Comment:
Would be helpful to indicate that this refers to active health checks, since Envoy has other forms (passive, EDS, ..).